module/zcommon/zfs_fletcher.c

```diff
@@ -245,7 +214,107 @@ fletcher_4_incremental_byteswap(const void *buf, uint64_t size,
 	ZIO_SET_CHECKSUM(zcp, a, b, c, d);
 }
 
+static const struct fletcher_4_calls fletcher_4_generic_calls = {
+	.init = fletcher_4_generic_init,
+	.compute = fletcher_4_generic,
+	.compute_byteswap = fletcher_4_generic_byteswap,
+	.name = "generic"
+};
+
+static const struct fletcher_4_calls *chosen = &fletcher_4_generic_calls;
```

Comment:
Adding "cycle" would be useful as well for `ztest`.  This causes the raidz code to use a different implementation each time to ensure good coverage and to make sure everything's as expected.  So in this cause it'd just alternate between "avx2" and "original".